game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -1428,98 +1429,22 @@ public static void triggerPlayerPropertyChange(final Set<TriggerAttachment> sati
-                    ChangeFactory.attachmentPropertyChange(attachment, newValue, property.getFirst(), clearFirst));
-              }
-              bridge.getHistoryWriter()
-                  .startEvent(MyFormatter.attachmentNameToText(t.getName()) + ": Setting " + property.getFirst()
-                      + (newValue.length() > 0 ? " to " + newValue : " cleared ") + " for "
-                      + t.getPlayerAttachmentName().getSecond() + " attached to " + player.getName());
-              break;
-            }
-            case "TechAttachment": {
-              final TechAttachment attachment = TechAttachment.get(player, t.getPlayerAttachmentName().getSecond());
-              if (newValue.equals(attachment.getRawPropertyString(property.getFirst()))) {
-                continue;
-              }
-              if (clearFirst && newValue.length() < 1) {
-                change.add(ChangeFactory.attachmentPropertyReset(attachment, property.getFirst()));
-              } else {
-                change.add(
-                    ChangeFactory.attachmentPropertyChange(attachment, newValue, property.getFirst(), clearFirst));
-              }
-              bridge.getHistoryWriter()
-                  .startEvent(MyFormatter.attachmentNameToText(t.getName()) + ": Setting " + property.getFirst()
-                      + (newValue.length() > 0 ? " to " + newValue : " cleared ") + " for "
-                      + t.getPlayerAttachmentName().getSecond() + " attached to " + player.getName());
-              break;
-            }
-            case "PoliticalActionAttachment": {
-              final PoliticalActionAttachment attachment =
-                  PoliticalActionAttachment.get(player, t.getPlayerAttachmentName().getSecond());
-              if (newValue.equals(attachment.getRawPropertyString(property.getFirst()))) {
-                continue;
-              }
-              if (clearFirst && newValue.length() < 1) {
-                change.add(ChangeFactory.attachmentPropertyReset(attachment, property.getFirst()));
-              } else {
-                change.add(
-                    ChangeFactory.attachmentPropertyChange(attachment, newValue, property.getFirst(), clearFirst));
-              }
-              bridge.getHistoryWriter()
-                  .startEvent(MyFormatter.attachmentNameToText(t.getName()) + ": Setting " + property.getFirst()
-                      + (newValue.length() > 0 ? " to " + newValue : " cleared ") + " for "
-                      + t.getPlayerAttachmentName().getSecond() + " attached to " + player.getName());
-              break;
-            }
-            case "UserActionAttachment": {
-              final UserActionAttachment attachment =
-                  UserActionAttachment.get(player, t.getPlayerAttachmentName().getSecond());
+          // TODO: Change to use 'Map.of()' factory method once using Java 9+.
+          // TODO add other attachment changes here if they attach to a player
```

Comment:
You could use guava's ImmutableMap builder if you want do do this a little bit cleaner ([ImmutableMap.Builder](https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/ImmutableMap.html#builder--))

Reply:
Fixed. I have also moved it out of the loops.